File: app/src/main/java/org/mifos/mobilewallet/invoice/InvoicePresenter.java
package org.mifos.mobilewallet.invoice;

import org.mifos.mobilewallet.core.BaseView;
import org.mifos.mobilewallet.core.UseCase;
import org.mifos.mobilewallet.core.UseCaseHandler;
import org.mifos.mobilewallet.home.HomeContract;
import org.mifos.mobilewallet.home.domain.usecase.FetchUserData;
import org.mifos.mobilewallet.invoice.domain.usecase.FetchPaymentMethods;

import javax.inject.Inject;

/**
 * Created by naman on 20/6/17.
 */

public class InvoicePresenter implements InvoiceContract.InvoicePresenter {

    private InvoiceContract.InvoiceView mInvoiceView;
    private final UseCaseHandler mUsecaseHandler;

    @Inject
    FetchPaymentMethods fetchPaymentMethods;

    @Inject
    public InvoicePresenter(UseCaseHandler useCaseHandler) {
        this.mUsecaseHandler = useCaseHandler;
    }

    @Override
    public void attachView(BaseView baseView) {
        mInvoiceView = (InvoiceContract.InvoiceView) baseView;
        mInvoiceView.setPresenter(this);
    }

    @Override
    public void fetchPaymentMethods() {
        mUsecaseHandler.execute(fetchPaymentMethods, null,
                new UseCase.UseCaseCallback<FetchPaymentMethods.ResponseValue>() {
                    @Override
                    public void onSuccess(FetchPaymentMethods.ResponseValue response) {
                        mInvoiceView.showPaymentMethods(response.getPaymentMethods());
                    }

                    @Override
                    public void onError(String message) {

                    }
                });
    }
}

File: mifospay/src/main/java/org/mifospay/invoice/presenter/InvoicePresenter.kt
package org.mifospay.invoice.presenter

import android.net.Uri
import org.mifospay.core.data.base.UseCase.UseCaseCallback
import org.mifospay.core.data.base.UseCaseHandler
import org.mifospay.core.data.domain.usecase.invoice.FetchInvoice
import org.mifospay.base.BaseView
import org.mifospay.core.datastore.PreferencesHelper
import org.mifospay.invoice.InvoiceContract
import org.mifospay.invoice.InvoiceContract.InvoiceView
import javax.inject.Inject

/**
 * Created by ankur on 07/June/2018
 */
class InvoicePresenter @Inject constructor(
    private val mUseCaseHandler: UseCaseHandler,
    private val mPreferencesHelper: PreferencesHelper,
    private val fetchInvoiceUseCase: FetchInvoice
) : InvoiceContract.InvoicePresenter {
    var mInvoiceView: InvoiceView? = null

    override fun attachView(baseView: BaseView<*>?) {
        mInvoiceView = baseView as InvoiceView?
        mInvoiceView!!.setPresenter(this)
    }

    override fun getInvoiceDetails(data: Uri?) {
        mUseCaseHandler.execute(fetchInvoiceUseCase, FetchInvoice.RequestValues(data),
            object : UseCaseCallback<FetchInvoice.ResponseValue> {
                override fun onSuccess(response: FetchInvoice.ResponseValue) {
                    mInvoiceView?.showInvoiceDetails(
                        response.invoices[0],
                        mPreferencesHelper.fullName + " "
                                + mPreferencesHelper.clientId, data.toString()
                    )
                }

                override fun onError(message: String) {
                    mInvoiceView!!.showToast(message)
                }
            })
    }
}